File: search_algorithms/binary_search/javascript/binary_search.js
const arr = [5, 3, 8, 4, 2]
const target = 9

function binarySearch(arr, target) {
  let low = 0;
  let high = arr.length - 1;

  while (low <= high) {
    let mid = Math.floor((low + high) / 2);
    if (arr[mid] < target) {
      low = mid + 1;
    } else if (arr[mid] > target) {
      high = mid - 1;
    } else {
      return mid;
    }
  }
  return -1
}

const result = binarySearch(arr, target)
console.log(result)

const outputElement = document.getElementById('output');
outputElement.innerHTML = '';
outputElement.innerHTML = `
  <p>Array: ${arr}</p>
  <p>Target: ${target}</p>
  <p>Find index or -1: ${result}</p>
  <p>For details, check the console.</p>
`;

File: search_algorithms/binary_search/python/binary_search.py
def binary_search(arr, target):
    """
    Perform a binary search on a sorted array to find the index of a target value.

    Parameters:
    arr (list): A sorted list of elements to search.
    target: The value to search for in the array.

    Returns:
    int: The index of the target value if found, otherwise -1.
    """

    low = 0
    high = len(arr) - 1

    while low <= high:
        mid = (low + high) // 2

        if arr[mid] < target:
            low = mid + 1
        elif arr[mid] > target:
            high = mid - 1
        else:
            return mid

    return -1


arr = [2, 3, 4, 10, 40]
x = 10
result = binary_search(arr, x)
if result != -1:
    print(f"Element is present at index {result}")
else:
    print("Element is not present in array")